agent/csi/mod_csi_docker.go

```diff
@@ -498,6 +515,22 @@ func (d *dockerBridge) Create(req *dvol.CreateRequest) error {
 
 func (d *dockerBridge) List() (*dvol.ListResponse, error) {
 
+	volMap := map[string]struct{}{}
+
+	// If the service is CSI-NFS then grab volumes from cache, as that's
+	// the only place they will be
+	if d.isCSINFS() {
+		d.byNameRWL.RLock()
+		defer d.byNameRWL.RUnlock()
+
+		for name, vi := range d.byName {
+			d.ctx.WithField("volume", vi.Id.Values).WithField(
+				"name", name).Debug(
+				"docker-csi-bridge: List: found volume from cache")
+			volMap[name] = struct{}{}
```

Comment:
Why is the cached value being set to an anonymous struct?

Reply:
This is just in the ListVolumes function, not the "cache" so to speak.  I really just need the name, and want to make sure they are unique later and therefore am using a map. I think you taught me that `struct{}` takes up no space. :)

Reply:
That's correct. Good trick!